database/sql/src/main/scala/cromwell/database/slick/MetadataSlickDatabase.scala

```diff
@@ -113,18 +132,21 @@ trait MetadataSlickDatabase extends MetadataSqlDatabase {
   }
 
   override def refreshMetadataSummaryEntries(metadataKey1: String, metadataKey2: String, metadataKey3: String,
-                                             metadataKey4: String, buildUpdatedSummary:
+                                             metadataKey4: String, metadataLabelKey: String, buildUpdatedSummary:
                                              (Option[WorkflowMetadataSummaryEntry], Seq[MetadataEntry]) =>
                                                WorkflowMetadataSummaryEntry)
                                             (implicit ec: ExecutionContext): Future[Long] = {
+    val likeMetadataLabelKey = metadataLabelKey + "%"
     val action = for {
       previousMetadataEntryIdOption <- getSummaryStatusEntryMaximumId(
         "WORKFLOW_METADATA_SUMMARY_ENTRY", "METADATA_ENTRY")
       previousMetadataEntryId = previousMetadataEntryIdOption.getOrElse(0L)
       metadataEntries <- dataAccess.metadataEntriesForIdGreaterThanOrEqual((
-        previousMetadataEntryId + 1L, metadataKey1, metadataKey2, metadataKey3, metadataKey4)).result
-      metadataByWorkflowUuid = metadataEntries.groupBy(_.workflowExecutionUuid)
+        previousMetadataEntryId + 1L, metadataKey1, metadataKey2, metadataKey3, metadataKey4, likeMetadataLabelKey)).result
+      metadataByWorkflowUuid = metadataEntries.filterNot(_.metadataKey.contains(metadataLabelKey)).groupBy(_.workflowExecutionUuid)
+      customLabelEntries = metadataEntries.filter(_.metadataKey.contains(metadataLabelKey))
```

Comment:
The naming of `metadataByWorkflowUuid` doesn't make it clear that this collection excludes labels.

You might build both collections with a single `partition`:

`(customLabelEntries, metadataWithoutLabels) = metadataEntries.partition(_.metadataKey.contains(metadataLabelKey))
`

Though you'd still need to run the `groupBy` on the non-label metadata.